model/metadata/metadata.go

```diff
@@ -99,3 +102,10 @@ func (m *Metadata) SetMinimal(fields common.MapStr) common.MapStr {
 	utility.Set(fields, "service", m.Service.MinimalFields())
 	return fields
 }
+
+func fetch(m common.MapStr, key string) string {
```

Comment:
`fetch` is usually associated with a remote call, this is just a get / lookup. and it could go in `utility` pkg

Reply:
Changed the name; I prefer to keep it local and unexported, we can revisit once this is needed more often.